lib/child_process.js

```diff
@@ -329,6 +329,49 @@ function normalizeSpawnArguments(file /*, args, options*/) {
   else if (options === null || typeof options !== 'object')
     throw new TypeError('"options" argument must be an object');
 
+  // Validate the cwd, if present.
+  if (options.cwd !== undefined &&
+      options.cwd !== null &&
+      !typeof options.cwd === 'string')
```

Comment:
Sounds like  string,  null and undefined are allowed.

On Aug 29, 2016 12:09 AM, "Jeremiah Senkpiel" <notifications@github.com>
wrote:

> In lib/child_process.js
> <https://github.com/nodejs/node/pull/8312#discussion_r76536916>:
>
> > @@ -329,6 +329,49 @@ function normalizeSpawnArguments(file /*, args, options*/) {
> >    else if (options === null || typeof options !== 'object')
> >      throw new TypeError('"options" argument must be an object');
> >
> > +  // Validate the cwd, if present.
> > +  if (options.cwd !== undefined &&
> > +      options.cwd !== null &&
> > +      !typeof options.cwd === 'string')
>
> Maybe a helper would be better? won't !typeof options.cwd === 'string'
> cover all these 3 cases?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/8312/files/b4a6558fcbcf6cde50ff29af2d3487896ad91c68#r76536916>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABjB841u9wl3B9jlsDIb7RdsDKU0iWTcks5qkdXdgaJpZM4Ju-4I>
> .
>